Missing dependency on PowerPCCodeGen leads to undefined references in tf_to_kernel

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): RHEL 7.4, PPC

**Describe the problem**

Running the tests with `bazel test ...` leads to undefined reference errors, see below log extract.
The reason is that `InitializeNativeTarget` is called which seemingly is a generated function and while the x86 dependency lib is linked via `@llvm-project//llvm:X86CodeGen`, the PPC equivalent is not.

The same problem may also affect ARM and arch64. Compare https://github.com/tensorflow/tensorflow/blob/15f4bda049539dd41c6dd9d0737d33da86cc32cf/tensorflow/compiler/aot/BUILD#L73-L83
with https://github.com/tensorflow/tensorflow/blob/15f4bda049539dd41c6dd9d0737d33da86cc32cf/tensorflow/compiler/mlir/tools/kernel_gen/BUILD#L122-L123

I think those lists should be pretty much identical.

**Provide the exact sequence of commands / steps that you executed before running into the problem**

`bazel test //tensorflow/...`

**Any other info / logs**

```
SUBCOMMAND: # //tensorflow/core/kernels/mlir_generated:invert_i32_kernel_cubin [action 'compile tensorflow/core/kernels/mlir_generated/invert_i32_kernel_cubin.sm_37.bin', configuration: 18b45a5a4afc1ae33c4d05
7c86644439e6ecd019473065d664167c1910aff23b, execution platform: @local_execution_config_platform//:platform]
...
bazel-out/ppc-opt/bin/tensorflow/compiler/mlir/tools/kernel_gen/_objs/tf_to_kernel/tf_to_kernel.o:tf_to_kernel.cc:function main: error: undefined reference to 'LLVMInitializePowerPCTargetInfo'
bazel-out/ppc-opt/bin/tensorflow/compiler/mlir/tools/kernel_gen/_objs/tf_to_kernel/tf_to_kernel.o:tf_to_kernel.cc:function main: error: undefined reference to 'LLVMInitializePowerPCTarget'
bazel-out/ppc-opt/bin/tensorflow/compiler/mlir/tools/kernel_gen/_objs/tf_to_kernel/tf_to_kernel.o:tf_to_kernel.cc:function main: error: undefined reference to 'LLVMInitializePowerPCTargetMC'
bazel-out/ppc-opt/bin/tensorflow/compiler/mlir/tools/kernel_gen/_objs/tf_to_kernel/tf_to_kernel.o:tf_to_kernel.cc:function main: error: undefined reference to 'LLVMInitializePowerPCAsmPrinter'
collect2: error: ld returned 1 exit status
```

--- a/tensorflow/compiler/mlir/tools/kernel_gen/BUILD
+++ b/tensorflow/compiler/mlir/tools/kernel_gen/BUILD
@@ -137,7 +137,6 @@ tf_cc_binary(
         "@llvm-project//mlir:ExecutionEngineUtils",
         "@llvm-project//mlir:Pass",
         "@llvm-project//mlir:TargetLLVMIR",
-    ],
     ] + if_llvm_system_z_available([
         "@llvm-project//llvm:SystemZCodeGen",  # fixdeps: keep
     ]) + if_llvm_aarch64_available([

--- a/tensorflow/compiler/mlir/tools/kernel_gen/BUILD
+++ b/tensorflow/compiler/mlir/tools/kernel_gen/BUILD
@@ -12,6 +12,7 @@ load(
     "@local_config_rocm//rocm:build_defs.bzl",
     "if_rocm_is_configured",
 )
+load("//tensorflow/core/platform:build_config.bzl", "if_llvm_aarch64_available", "if_llvm_system_z_available")
 
 package(
     default_visibility = [":friends"],
@@ -125,8 +126,10 @@ tf_cc_binary(
         "//tensorflow/stream_executor/lib",
         "@com_google_absl//absl/strings",
         "@llvm-project//llvm:Analysis",
+        "@llvm-project//llvm:ARMCodeGen",  # fixdeps: keep
         "@llvm-project//llvm:CodeGen",
         "@llvm-project//llvm:Core",
+        "@llvm-project//llvm:PowerPCCodeGen",  # fixdeps: keep
         "@llvm-project//llvm:Support",
         "@llvm-project//llvm:Target",
         "@llvm-project//llvm:X86CodeGen",  # fixdeps: keep
@@ -135,6 +138,11 @@ tf_cc_binary(
         "@llvm-project//mlir:Pass",
         "@llvm-project//mlir:TargetLLVMIR",
     ],
+    ] + if_llvm_system_z_available([
+        "@llvm-project//llvm:SystemZCodeGen",  # fixdeps: keep
+    ]) + if_llvm_aarch64_available([
+        "@llvm-project//llvm:AArch64CodeGen",  # fixdeps: keep
+    ]),
 )
 
 tf_cc_binary(

